ui/app/services/flags.ts

```diff
@@ -24,22 +25,37 @@ export default class flagsService extends Service {
   @service declare readonly version: VersionService;
   @service declare readonly store: StoreService;
 
-  @tracked flags: string[] = [];
   @tracked activatedFlags: string[] = [];
+  @tracked featureFlags: string[] = [];
 
-  setFeatureFlags(flags: string[]) {
-    this.flags = flags;
+  get isHvdManaged(): boolean {
+    return this.featureFlags?.includes(FLAGS.vaultCloudNamespace);
   }
 
-  get managedNamespaceRoot() {
-    if (this.flags && this.flags.includes(FLAGS.vaultCloudNamespace)) {
-      return 'admin';
+  get hvdManagedNamespaceRoot(): string | null {
+    return this.isHvdManaged ? 'admin' : null;
+  }
+
+  getFeatureFlags = keepLatestTask(async () => {
+    try {
+      const result = await fetch('/v1/sys/internal/ui/feature-flags', {
+        method: 'GET',
+      });
+
+      if (result.status === 200) {
+        const body = await result.json();
+        this.featureFlags = body.feature_flags || [];
+      }
+    } catch (error) {
+      if (DEBUG) console.error(error); // eslint-disable-line no-console
     }
-    return null;
+  });
+
+  fetchFeatureFlags() {
+    return this.getFeatureFlags.perform();
```

Comment:
Do we want to add any kind of caching, so that we don't refetch the flags if they've already been retrieved?

Reply:
Another conservative approach on my end. The previous behavior did not have a caching check and because we use this response for things outside Secrets Sync I didn't want to modify the behavior. 😬